### DISCUSSION TITLE
Reporting of cAPI active

### DISCUSSION THREAD

**Discussion by ACyprus** [EDCD/EDDI#2344] [EDDI VoiceAttack Q&A] [answered]
Hey o7

This might be better placed in a new Issue but I wanted to confirm my understanding before creating one.

## Context

- I'd like to create a VA command that confirms EDDI's Frontier API status is connected, but I can't seem to ever find when the corresponding variable is `true`.

## What I've tried

- using the `{BOOL:cAPI active}` token in a text variable and sending that to the speech responder
- using the `cAPI active` variable in a `Boolean Compare` action
- using `icao active` as a control test in both cases above (this works properly)
- running this from the `((EDDI VA Initialized))` event and a manually executed command

## Testing steps

- Start VA
- open EDDI config UI
- EDDI UI confirms the API status is `Connected`
- run test command
  - `cAPI active` evaluates to `Not Set`
---
- disconnect the API via the EDDI config UI
- EDDI UI confirms the API status is `Not Connected`
- run test command
  - `cAPI active` evaluates to `False`
---
- restart VA
- run test command
  - `cAPI active` evaluates to `Not Set`
---
- quit VA and open EDDI proper
- reconnect the Frontier API
- EDDI UI confirms the API status is `Connected` 
- close EDDI and restart VA
- run test command
  - `cAPI active` evaluates to `Not Set`

## Questions

- am I using the variable in the wrong way?
- am I using the wrong variable altogether?

Thanks! Hope I'm not wasting your time with a silly mistake.

**Comment by Tkael** [comment, marked as answer]
Reviewing your comments and the code, I suspect the value may remain unset in VoiceAttack until there is a change in the Frontier API state. Since you cannot connect except while running in standalone mode, this would definitely be an issue. The only detectable state would be a disconnect.
